subprojects/analysis-kotlin-api/src/testFixtures/kotlin/org/jetbrains/dokka/analysis/test/api/analysis/TestProjectAnalyzer.kt

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param persistentDirectory a directory that will be used to generate the input test files into.
+     *                            It must be available during the test run, especially if services are used,
+     *                            otherwise parts of Dokka might not work as expected. Can be safely deleted
+     *                            at the end of the test after all asserts have been run.
+     *
+     * @see [TestProject.useServices] for a user-friendly way to call it
+     */
+    fun analyze(
+        testProject: TestProject,
+        persistentDirectory: File
+    ): Pair<TestAnalysisServices, TestAnalysisContext> {
+        val (dokkaConfiguration, dokkaContext) = testProject.initialize(outputDirectory = persistentDirectory)
+        val analysisServices = createTestAnalysisServices(dokkaContext)
+        val testAnalysisContext = TestAnalysisContext(
+            context = dokkaContext,
+            configuration = dokkaConfiguration,
+            module = generateDocumentableModel(dokkaContext)
+        )
+        return analysisServices to testAnalysisContext
+    }
+
+    /**
+     * Prepares this [TestProject] for analysis by creating
+     * the test files, setting up context and configuration.
+     */
+    private fun TestProject.initialize(outputDirectory: File): Pair<DokkaConfiguration, DokkaContext> {
+        analysisLogger.progress("Initializing and verifying project $this")
+        this.verify()
+        require(outputDirectory.exists() && outputDirectory.isDirectory) {
+            "outputDirectory has to exist and be a directory: $outputDirectory"
+        }
+        this.initializeTestFiles(relativeToDir = outputDirectory)
+
+        analysisLogger.progress("Creating configuration and context")
+        val testDokkaConfiguration = this.getConfiguration()
+        val dokkaConfiguration = testDokkaConfiguration.toDokkaConfiguration(projectDir = outputDirectory).also {
+            it.verify()
+        }
+        return dokkaConfiguration to createContext(dokkaConfiguration)
+    }
+
+    /**
+     * Takes the virtual [TestDataFile] of this [TestProject] and creates
+     * the real files relative to the [relativeToDir] param.
+     */
+    private fun TestProject.initializeTestFiles(relativeToDir: File) {
+        analysisLogger.progress("Initializing test files relative to the \"$relativeToDir\" directory")
+
+        this.getTestData().getFiles().forEach {
+            val testDataFile = relativeToDir.resolve(it.pathFromProjectRoot.removePrefix("/"))
+            try {
+                testDataFile.parentFile.mkdirs()
+            } catch (e: Exception) {
+                // the IOException thrown from `mkdirs()` has no details and thus is more difficult to debug.
+                throw IllegalStateException("Unable to create dirs \"${testDataFile.parentFile}\"", e)
+            }
+
+            analysisLogger.debug("Creating \"${testDataFile.absoluteFile}\"")
```

Comment:
Nitpicking: Do you mean  `testDataFile.absolutePath`?

Reply:
Probably, yeah. Fixed it